crates/core/src/client/helpers.rs

```diff
@@ -138,18 +140,16 @@ pub fn decode_eth_send_transaction_return<T: JsonRpcTransport>(
     // fail is Kakarot API changes)
     // Declare Vec of Result
     let mut segmented_result: Vec<FeltOrFeltArray> = Vec::new();
-    let mut tmp_array_len: FieldElement = *call_result.first().ok_or_else(|| {
-        EthApiError::OtherError(anyhow::anyhow!("Cannot parse and decode return arguments of Kakarot call",))
-    })?;
+    let mut tmp_array_len: FieldElement = *call_result
```

Comment:
same here

Reply:
just proposed to remove this function in any case https://github.com/kkrt-labs/kakarot-rpc/pull/241